### PR TITLE
updated Scan view to align w/ security document, plus readability polish

### PR DESCRIPTION
**Pull request by mertbagt**
changed allowUpdate to be based on user.office_code instead of user.can_update_status_for

### CODE REVIEW

**Review by ethanstrominger — CHANGES_REQUESTED** [review]

File: src/pages/scanOrder.js [hunk 1]
```diff
@@ -161,42 +161,30 @@ const ScanOrder = (props) => {
...
-    }
-    if (user.can_update_status_for === "ALL") {
-      allowHOSS = "yes";
-      allowAOC = "yes";
-      allowMAIL = "yes";
-      allowSTATE = "yes";
-    }
-  }
+  const calculateAllowUpdate = () => {
+    if (nextStatus.permission === "FED-HOSS" && user.office_code === "FED-HOSS")
```
[the remark this answers is not in the document]
Comment (mertbagt) [reply src/pages/scanOrder.js]: done

File: src/pages/scanOrder.js [hunk 2]
```diff
@@ -161,42 +161,30 @@ const ScanOrder = (props) => {
...
-    allowUpdate = "yes";
-  if (nextStatus.permission === "STATE" && allowSTATE === "yes")
-    allowUpdate = "yes";
+    if (
+      nextStatus.permission === "STATE" &&
+      user.office_code === order.home_office_code
+    )
+      allowUpdate = "yes";
+
+    if (user.office_code === "FED-ADMIN") allowUpdate = "yes";
```
[the remark this answers is not in the document]
Comment (mertbagt) [reply src/pages/scanOrder.js]: done

File: src/pages/scanOrder.js [hunk 3]
```diff
@@ -161,42 +161,30 @@ const ScanOrder = (props) => {
...
+  };
+
+  if (user) {
```
[the remark this answers is not in the document]
Comment (mertbagt) [reply src/pages/scanOrder.js]: done

File: src/pages/scanOrder.js [hunk 4]
```diff
@@ -161,42 +161,30 @@ const ScanOrder = (props) => {
...
-      allowSTATE = "yes";
-    }
-    if (user.can_update_status_for === "ALL") {
-      allowHOSS = "yes";
-      allowAOC = "yes";
-      allowMAIL = "yes";
-      allowSTATE = "yes";
-    }
-  }
+  const calculateAllowUpdate = () => {
```
[the remark this answers is not in the document]
Comment (mertbagt) [reply src/pages/scanOrder.js]: - [X] allowUpdate refactored 
- [ ] refactor lifeCycle
- [ ] rename lifeCycle

**Comment by mertbagt** [comment]
image

**Pull request merged**